src/main/java/com/openshift/internal/restclient/model/KubernetesResource.java

```diff
@@ -105,11 +105,10 @@ public void refresh(){
 		this.node = ModelNode.fromJSONString(client.get(getKind(), getName(), getNamespace()).toString());
 	}
 	
-	// TODO Pretty certain this should be protected
 	@Override
 	public ResourceKind getKind(){
-		if(node.has("kind")){
-			return ResourceKind.valueOf(node.get("kind").asString());
+		if(node.has(ResourcePropertyKeys.KIND)){
```

Comment:
@adietish this works for both "labels" and "kind" but technically is in correct. ResourcePropertyKeys is really a key value that should never change to a path defined in ResourcePropertiesRegistry.  This path is versioned and gives as the value at defined by the path.  This class has a number of helper methods that retrieve values based on keys and the direct use of node here is probably legacy from when it was first implemented.  We should consider using the helper methods.